browser-test/src/support/admin_programs.ts

```diff
@@ -148,6 +148,35 @@ export class AdminPrograms {
     }
   }
 
+  async editProgramBlockWithOptional(programName: string, blockDescription = 'screen description', questionNames:
+  string[], optionalQuestionName: string) {
+    await this.gotoDraftProgramEditPage(programName);
+
+    await this.page.click('text=Manage Questions');
+    await waitForPageJsLoad(this.page);
+    await this.expectProgramBlockEditPage(programName);
+    await waitForPageJsLoad(this.page);
+
+    await this.page.click('#block-description-modal-button');
+    await waitForPageJsLoad(this.page);
+    await this.page.fill('textarea', blockDescription);
+    await this.page.click('#update-block-button');
+    await waitForPageJsLoad(this.page);
+
+    // Add the optional question
+    await this.page.click(`button:text("${optionalQuestionName}")`);
+    await waitForPageJsLoad(this.page);
+    // Only allow one optional question per block; this selector will always toggle the first optional button.  It
+    // cannot tell the difference between multiple option buttons
+    await this.page.click(`:is(button:has-text("optional"))`);

```

Comment:
This meets the need for now; I've documented the limitation so that if someone attempts to use it in the future they will know and can expand it appropriately.  Feel free to reopen if you think it's critical that this be more specific now; currently it will always select the optional button